legacy/application/models/StreamSetting.php

```diff
@@ -55,16 +55,28 @@ public static function getOutput($key, $add_prefix = false)
                 $prefix . 'pass' => $output['source_password'] ?? '',
                 $prefix . 'admin_user' => $output['admin_user'] ?? 'admin',
                 $prefix . 'admin_pass' => $output['admin_password'] ?? '',
-                $prefix . 'channels' => $output['audio']['channels'] ?? 'stereo',
-                $prefix . 'bitrate' => $output['audio']['bitrate'] ?? 128,
-                $prefix . 'type' => $output['audio']['format'],
                 $prefix . 'name' => $output['name'] ?? '',
                 $prefix . 'description' => $output['description'] ?? '',
                 $prefix . 'genre' => $output['genre'] ?? '',
                 $prefix . 'url' => $output['website'] ?? '',
                 $prefix . 'mobile' => $output['mobile'] ?? 'false',
                 // $prefix . 'liquidsoap_error' => 'waiting',
             ];
+            if (array_key_exists('audio', $output)) {
+                $result = array_merge($result, [
+                    $prefix . 'channels' => $output['audio']['channels'] ?? 'stereo',
+                    $prefix . 'bitrate' => $output['audio']['bitrate'] ?? 128,
+                    $prefix . 'type' => $output['audio']['format'],
+                ]);
+            } elseif ($output['kind'] == 'hls') {
+                // HLS : set web server host and port
+                $result = array_merge($result, [
+                    $prefix . 'port' => $_SERVER['SERVER_PORT'],
+                    $prefix . 'host' => $_SERVER['SERVER_NAME'],
+                    $prefix . 'type' => 'hls',
+                    $prefix . 'bitrate' => '',
+                ]);
```

Comment:
Hmm, I think we prefer to use the config public url and not the webserver configs.

Reply:
in the present configuration, you can use both as with other outputs:
- default is webserver
- if public url is provided  it's use instead

Reply:
The rest of the project doesn't rely on the web server configuration, this would diverge from the current expectation.

Reply:
so you tell me to add settings for web server host/port?

Reply:
We use the `Config::get('general.public_url')` (with the _raw key you get access to the URL object (use League\Uri\Uri;).

https://github.com/libretime/libretime/blob/e2cfbf4c038f28874a206df5805f04f69a40647b/legacy/application/models/StreamSetting.php#L70-L81

Reply:
But I think we create an new function and not use the backward compatible and complex Application_Model_StreamConfig::getOutput function.